### DISCUSSION TITLE
HomeAssistant Integration

### DISCUSSION THREAD

**Discussion by sgayda2** [PlummersSoftwareLLC/NightDriverStrip#627] [Ideas]
It would be cool to have home assistant integration.

This would allow someone to adjust an animation/color/strip based on some triggers within your home assistant automations. I believe that WLED has this kind of support, and I think it would be cool to add something similar to NightDriver

**Comment by robertlipe** [comment]
There's a JSON API
<https://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/REST_API.md>
that exposes darned near everything we can do.
YOu might not be able to light up a number of LEDS = the db reception when
your cell phone gets closer,or trigger OTAs as described by the WLED
integration <https://www.home-assistant.io/integrations/wled/> but I'd
expect basic turning things off and back on again (Hi, Moss!) should be
pretty straightforward for someone with knowledge of that half that's
suitably motivated. Just attach some curl-like
<https://community.home-assistant.io/t/execute-curl-command-from-an-automation/525604>
POST and GET action driven from your triggers in HA to make the lights go
brrr.

I don't perceive there to be a huge amount of actual programmer-level
skills needed, just some basic scripting style tasks, but I don't do HA.

If you can drive the HA side, one of us here can help you fill in any
needed gaps in this code.  At a glance, we might want to do something more
friendly like a "power on/off" than a "change settings/brightness to
zero/255" sort of hook, but that's not particularly hard.

Are you willing to help drive it? That's what's been missing so far from
what I can tell.

**Reply by sgayda2** [reply]
Unfortunately i don't have the spare time for this, but I was just trying to understand the state of things. Seems like its not far off if I end up with some time+motivation.

**Comment by robertlipe** [comment]
Open source pretty much always relies upon someone with the right skills
(which can be outsourced - contract programmers and folks needing starter
projects exist) and motivation to make things happen.

I'd say your assessment is right. It's not "in the box", but it's probably
not THAT difficult in total. I'm comfortable with this side and would even
be willing to help someone else tackle it. I just don't do HA.

I recently helped a friend with a WLED deployment.  He's a geek, but not a
programmer and he did the HA integration without me. (This did not make me
*un*happy. :-) ) He is quite pleased with the results.  If WLED is a better
fit, it's hard to suggest NOT using it. It's a well done project and uses
many of the same underpinnings that we do here.

You have the blessing/curse of multiple, free, customizable options. Nice
problems to have!

Good luck,
Blinky Master Robert
